pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -40,14 +48,45 @@ func nameAndTag(id string) (name string, tag string, err error) {
 		name = segments[0]
 		tag = segments[1]
 		if len(name) == 0 || len(tag) == 0 {
-			err = errors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
+			err = kapierrors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
 		}
 	default:
-		err = errors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
+		err = kapierrors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
 	}
 	return
 }
 
+func (r *REST) List(ctx kapi.Context, label labels.Selector, field fields.Selector) (runtime.Object, error) {
+	imageStreams, err := r.imageStreamRegistry.ListImageStreams(ctx, labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	matcher := MatchImageStreamTag(label, field)
```

Comment:
This isn't correct because it doesn't return images as well.  I'm not sure it's worth implementing if the image metadata is empty.  That means we'll do an N+1 join here.